### PR TITLE
Show the subscription fields when the product type is simple subscription

### LINKED ISSUE #75: Shipping and Advanced settings are missing from Simple Subscription products
## Describe the bug

When creating a Simple Subscription product:

- the "One time shipping" option under **Product Data > Shipping** and; 
- the "Limit Subscription" option under **Product Data > Advanced**, 
are hidden. 

Here's a video depicting this behavior: 

https://user-images.githubusercontent.com/25955483/145987558-78e91c84-9fdd-42a7-97d3-dcb75d6d8d7c.mp4

## To Reproduce

1. Go to the Products tab
2. Create a new Simple Subscription product.
3. Navigate to either **Product Data > Shipping** or **Product Data > Advanced** and check if Subscription specific settings show up there.

### Expected behavior
The expected behavior is that "One time shipping" and "Limit Subscription" settings would show up for Simple Subscription products.

### Actual behavior
The actual behavior is that these settings only show up for Variable Subscription products but not for Simple Subscription ones. 

## Product impact

- [x] Does this issue affect WooCommerce Subscriptions? Yes 4288-gh-woocommerce/woocommerce-subscriptions
- [x] Does this issue affect WooCommerce Payments? Yes https://github.com/Automattic/woocommerce-payments/issues/3537

This issue can have a major product impact as many of the Subscription-specific settings do not show up for Simple Subscription products. So, customers cannot use many of the advertised features. 

In addition to that, customers who transitioned from WooCommerce Subscriptions to WooCommerce Payments will not be able to edit the values of these settings. 

## Additional context

This issue was introduced via: https://github.com/Automattic/woocommerce-subscriptions-core/commit/da32cf1d9073b40ca32d34a9381a9a6d2288aeb3#diff-0dec7a408ff08c7c6200cd2b92cdc10704d7b68c3066c70f9e3c89ebadd6a7eb

The aim of that commit was to fix an issue with Subscription-related fields incorrectly showing up/getting hidden when adding new variations to a product. 

To fix this, this commit sets all Subscription-related fields to hidden for all product types. Then, if the current product is a Variable Subscription product, these fields get shown. However, there isn't any code for what should happen for Simple Subscription products -- therefore, these fields remain hidden for Simple Subscription products.

### LINKED ISSUE #3537: Shipping and Advanced settings missing for `Simple Subscription` products
**This is a placeholder for tracking how this issue affects WCPay. Please use the primary issue in subscriptions-core repository for all discussion:** https://github.com/Automattic/woocommerce-subscriptions-core/issues/75 

### Describe the bug
Subscription products have additional options in the product edit metabox, specifically:

- `One time shipping` option under `Product Data > Shipping` 
-  `Limit Subscription` option under `Product Data > Advanced`

These options are not available. 

The markup has been hidden due to changes in this PR: https://github.com/Automattic/woocommerce-subscriptions-core/pull/33

Thanks to @jimjasson for finding this!

### PR DESCRIPTION
**Pull request by aprea**
Fixes #75
Fixes https://github.com/Automattic/woocommerce-payments/issues/3537

## Description

Fixes an issue introduced in da32cf1d9073b40ca32d34a9381a9a6d2288aeb3 when creating a Simple Subscription product:

- the "One time shipping" option under **Product Data > Shipping** and; 
- the "Limit Subscription" option under **Product Data > Advanced**, 
are hidden.

## How to test this PR

**Note:** these steps are largely copied from 4215-gh-woocommerce/woocommerce-subscriptions

1. Add the following code to your site somewhere (e.g. as a mu-plugin). 
1. Go to **Products > Add new**. 
1. Select the **New Variable**.
2. Go to the **Attributes** tab and add an attribute with attribute values.
3. Check the **'Used for variations'** checkbox.
4. Go to the **Variations** tab.
5. Click **go**. 
6. Expand the variation level settings.
7. Observe, there shouldn't be any subscription-related fields shown. https://d.pr/i/WIu1jH
8. Change the product type to **"Simple subscription"**.
9. Click the **"Shipping"** tab.
10. Observe, the **"One time shipping"** option is available https://d.pr/i/FaARPN
11. Click the **"Advanced"** tab.
12. Observe, the **"Limit subscription"** option is available https://d.pr/i/Ga30Bg
13. Change the product type to **"Variable subscription"**.
14. Click the **"Shipping"** tab.
15. Observe, the **"One time shipping"** option is available https://d.pr/i/z6jv9H
16. Click the **"Advanced"** tab.
17. Observe, the **"Limit subscription"** option is available https://d.pr/i/5vuRVE
18. Change the product type to **"Simple product"**.
19. Click the **"Shipping"** tab.
20. Observe, the **"One time shipping"** option is not available.
21. Click the **"Advanced"** tab.
22. Observe, the **"Limit subscription"** option is not available.

```php
/**
 * Add option to product type selector.
 */
function kia_new_variable_product_type( $product_types ){
    $product_types[ 'new-variable' ] = 'New Variable';
    return $product_types;
}
add_filter( 'product_type_selector', 'kia_new_variable_product_type' );

/**
 * Define new product class.
 */
function kia_create_new_product_class(){
    class WC_Product_New_Variable extends WC_Product_Variable {


        public function __construct( $product ) {

            $this->product_type = 'new-variable';
            parent::__construct( $product );

        }

        public function get_type() {
            return 'new-variable'; // so you can use $product = wc_get_product(); $product->get_type()
        }

    }
}
add_action( 'woocommerce_loaded', 'kia_create_new_product_class' );

/**
 * Custom JS for metaboxes.
 */
function kia_producttype_custom_js() {

	wp_add_inline_script( 'wc-admin-product-meta-boxes', '
		jQuery( function( $ ) {
				$( ".variations_tab" ).addClass( "show_if_new-variable" );

				$( document.body ).on( "woocommerce_added_attribute", function() {
					$( ".enable_variation" ).addClass( "show_if_new-variable" );

					if ( "new-variable" === $( "select#product-type" ).val() ) {
						$( ".enable_variation" ).show();
					}
				});
			} );
	', 'before' );
}
add_action( 'admin_enqueue_scripts', 'kia_producttype_custom_js', 99 );

/**
 * Use existing data store.
 */
function kia_new_variable_data_store( $stores ) {
    $stores['product-new-variable'] = 'WC_Product_Variable_Data_Store_CPT';
    return $stores;
}
add_filter( 'woocommerce_data_stores', 'kia_new_variable_data_store' );
```

## Product impact

- [ ] Added changelog entry (or does not apply)
- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref

### CODE REVIEW

**Comment by shendy-a8c** [comment]
I think your change works to solve https://github.com/Automattic/woocommerce-subscriptions-core/issues/75 but maybe not for pdjTHR-m5-p2#comment-571 (where element with both CSS classes `hide_if_subscription` and `hide_if_variable-subscription` should be hidden when product type is `subscription`).

I think if we change [these 2 lines](https://github.com/Automattic/woocommerce-subscriptions-core/commit/da32cf1d9073b40ca32d34a9381a9a6d2288aeb3#diff-0dec7a408ff08c7c6200cd2b92cdc10704d7b68c3066c70f9e3c89ebadd6a7ebR48-R49) to:
```
			$( '.hide_if_variable-subscription' )
				.filter( function () {
					return 'variable-subscription' !== $( 'select#product-type' ).val() && ! $(this).hasClass( 'hide_if_' + $( 'select#product-type' ).val() );
				} )
				.show();
			$( '.show_if_variable-subscription' )
				.filter( function () {
					return 'variable-subscription' !== $( 'select#product-type' ).val() && ! $(this).hasClass( 'show_if_' + $( 'select#product-type' ).val() );
				} )
				.hide();
```
and revert your change, it will fix both https://github.com/Automattic/woocommerce-subscriptions-core/issues/75 and pdjTHR-m5-p2#comment-571, and won't break 4192-gh-woocommerce/woocommerce-subscription, maybe. What do you think?

I need to test more but my brain is melting thinking of all the edge cases. Will continue again tomorrow.

My train of thought is not to assume the element only has 1 `hide_if_` or `show_if_` CSS class, so avoiding counterintuitively `show()` / `hide()`-ing if it has `hide_if_{product_type}` or `show_if_{product_type}` respectively.

File: assets/js/admin/admin.js [hunk 1]
```diff
@@ -74,6 +74,10 @@ jQuery( function ( $ ) {
 					$.showOrHideStockFields();
 				}
 
+				if ( 'subscription' === $( 'select#product-type' ).val() ) {
+					$( '.show_if_subscription' ).show();
```
Comment (shendy-a8c) [inline assets/js/admin/admin.js:78]:
```suggestion
					$( '.show_if_subscription' ).show();
					$( '.hide_if_subscription' ).hide();
```

**Review by shendy-a8c — COMMENTED** [review]
Adding this 1 line to your change would also fix pdjTHR-m5-p2#comment-571, I think.

File: assets/js/admin/admin.js (hunk 1 above)
Comment (aprea) [reply assets/js/admin/admin.js:78]: Thanks @shendy-a8c, I have implemented this change!

**Review by shendy-a8c — APPROVED** [review]
Looks good. I have tested per your instructions and also manually add CSS class `hide_if_subscription` and `hide_if_variable-subscription` to a certain element and that element is hidden correctly when product type is subscription or variable-subscription.
